src/base/UnorderedNonce.sol

```diff
@@ -1,11 +1,11 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import {IUnorderedNonce} from "../interfaces/IUnorderedNonce.sol";
+
 /// @title Unordered Nonce
 /// @notice Contract state and methods for using unordered nonces in signatures
-contract UnorderedNonce {
-    error NonceAlreadyUsed();
-
+contract UnorderedNonce is IUnorderedNonce {
     /// @notice mapping of nonces consumed by each address, where a nonce is a single bit on the 256-bit bitmap
     /// @dev word is at most type(uint248).max
     mapping(address owner => mapping(uint256 word => uint256 bitmap)) public nonces;
```

Comment:
cant this natspec go in the interface too? mayve inheritdoc gets confused with mappings

Reply:
could, but then i will need to define another function for `nonces`

Reply:
& its a bit odd bc in the interface it will be declared as external but here it is a public variable

Reply:
I made this change so you could see, but lmk if you want me to go back to this. idc either way tbh

Reply:
yeah i like it! its normal that public functions have to be external in interfaces - just good for people to be able to use the interface to access all public information

Reply:
just need to add ` /// @inheritdoc IUnorderedNonce` here now